### PR TITLE
Wip unified cdc 2

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java [hunk 1]
```diff
@@ -118,20 +134,14 @@ protected void leaderStart() {
...
 
-          SchemaAndTable schemaAndTable = new SchemaAndTable(tableMapEvent.getDatabase(), tableMapEvent.getTable());
-
-//          if (!cdcMonitoringTableId.isPresent() &&
-//                  schemaAndTable.equals(new SchemaAndTable(new EventuateSchema().getEventuateDatabaseSchema(), "cdc_monitoring"))) {
-//            cdcMonitoringTableId = Optional.of(tableMapEvent.getTableId());
-//            break;
-//          }
+          cdcMonitoringTableId = Optional.empty();
+          tableMapEventByTableId.clear();
```
Comment (cer) [inline new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java]: Hmmm.... are you sure you should clear it.
This appears to suggest that its a map with at most one entry?

File: new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java [hunk 2]
```diff
@@ -207,11 +218,10 @@ private void handleWriteRowsEvent(Event event, Optional<BinlogFileOffset> starti
     BinlogFileOffset binlogFileOffset = new BinlogFileOffset(binlogFilename, offset);
     logger.info("mysql binlog client got event with offset {}", binlogFileOffset);
 
-    /*if (cdcMonitoringTableId.map(id -> id.equals(eventData.getTableId())).orElse(false)) {
-      onMonitoringEventReceived();
+    if (cdcMonitoringTableId.map(id -> id.equals(eventData.getTableId())).orElse(false)) {
```
Comment (cer) [inline new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java]: `cdcMonitoringTableId.map(id -> id.equals(eventData.getTableId())).orElse(false)` occurs twice. Extract into a method `isCdcMonitor....(...)`

File: new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java [hunk 3]
```diff
@@ -230,21 +240,21 @@ private void handleWriteRowsEvent(Event event, Optional<BinlogFileOffset> starti
...
+
+    if (eventData == null) {
+      return;
+    }
+
+    if (cdcMonitoringTableId.map(id -> id.equals(eventData.getTableId())).orElse(false)) {
+      onMonitoringEventReceived(timestampExtractor.extract(MONITORING_SCHEMA_AND_TABLE, eventData));
+    }
+
+    offset = ((EventHeaderV4) event.getHeader()).getPosition();
```
Comment (cer) [inline new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java]: These 3 lines are important and occur more than once in this class. Extract into method

File: new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java (hunk 1 above)
Comment (dartartem) [reply new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java]: From the docs, TABLE_MAP event precedes each row operation event.
So required id is always present in map when write/update event is received. So it should work.
In current case it solves the issue with changed ids when mysql is restarted.
I am not sure that there is better solution.

File: new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java (hunk 2 above)
Comment (dartartem) [reply new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java]: ok

File: new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java (hunk 3 above)
Comment (dartartem) [reply new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java]: ok

**Pull request merged**